pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -710,17 +723,20 @@ func decreaseAndCheckMaxUnavailable(maxUnavailable *int) bool {
 	  reserveOrdinals:
 	  - 1
       - 3
+      Spec.Ordinals.Start: 2
 */
-// return replicaCount=6, reserveOrdinals={1, 3}
-
+// result is maxReplicaIdx = 5, reserveOrdinals = {1, 3}
+// replicas[5] stores [replica-2, nil(reserveOrdinal 3), replica-4, replica-5, replica-6]
+// todo: maybe we should remove ineffective reserveOrdinals in webhook, reserveOrdinals = {3}
 func getStatefulSetReplicasRange(set *appsv1beta1.StatefulSet) (int, sets.Int) {
```

Comment:
replicas range is not so well defined if start ordinal is non-zero, maybe we could return start, end and reserved ordinals